src/datachain/lib/file.py

```diff
@@ -190,6 +190,24 @@ def __init__(self, **kwargs):
         self._catalog = None
         self._caching_enabled: bool = False
 
+    @classmethod
+    def upload(
+        cls, path: str, data: bytes, catalog: Optional["Catalog"] = None
+    ) -> "File":
+        if catalog is None:
+            from datachain.catalog.loader import get_catalog
+
+            catalog = get_catalog()
+
+        parts = path.split("/")
+        parent = "/".join(parts[:-1])
+        name = parts[-1]
```

Comment:
Please use `posixpath.split()` instead of doing this by yourself.

Reply:
There are examples of `.split("/")` and `"/".join` throughout the codebase.

Reply:
(don't know the exact context here)

> There are examples of .split("/") and "/".join throughout the codebase.

in some case it is safe (cloud URLs), in some cases it's not (e.g. local file system URL - it can be `/` or `\`)